tests/main/uc20-snap-recovery-encrypt/task.yaml

```diff
@@ -92,15 +97,18 @@ execute: |
     umount /run/mnt/ubuntu-data
     cryptsetup close /dev/mapper/ubuntu-data
 
-    echo "ensure that we can open the encrypted device using the key"
+    echo "Ensure that we can open the encrypted device using the key"
     mkdir -p ./mnt
     cryptsetup open --key-file keyfile "${LOOP}p4" test-udata
     mount /dev/mapper/test-udata ./mnt
+    echo "Check that the recovery key was stored"
+    test "$(stat --printf="%s" ./mnt/system-data/recovery-key)" = 16
+    cp ./mnt/system-data/recovery-key .
     umount ./mnt
     cryptsetup close /dev/mapper/test-udata
 
-    # debug message to see if the udev database is correctly updated
-    udevadm info --query=property "${LOOP}p2" | grep ID_FS_TYPE ||:
-
-    # XXX: can't add more partitions after an encrypted partition is created
-    # (gadget compatibility tests fail)
+    echo "Ensure that we can open the encrypted device using the recovery key"
+    cryptsetup open --key-file recovery-key "${LOOP}p4" test-recovery
+    mount /dev/mapper/test-recovery ./mnt
```

Comment:
Tabs vs spaces?

Reply:
Apparently we're already using spaces everywhere in this file.